doc/build-versioned-docs.sh

```diff
@@ -76,7 +76,9 @@ END
 
     # Restore branch as it was to avoid conflicts
     git restore source/_templates
-    rm -rf source/ref-api
+    git restore source/_templates/autosummary || rm -rf source/_templates/autosummary
```

Comment:
LGTM! Could you just explain behind this line?

Reply:
This line tries to restore using git if anything was changed on those files compared to the branch it is in and if it is not able to (because those files do not exist on that branch), it deletes them.
For a bit of context, this is needed because the versioning is done by checking out the different tagged versions, and so the current checked out branch shouldn't be changed before checking out the following one, otherwise it won't work.